### PR TITLE
Footer view component [DS-11]

### PR DESCRIPTION
**Pull request by davidrapson**
## Background

https://citizensadvice.atlassian.net/browse/DS-11

## Changes

Adds view component version of the footer component.

Uses [slots](https://viewcomponent.org/guide/slots.html) to handle link columns. Limits to four columns.

**TODO:**

- [x] Add component previews to help define slots API
- [x] Add component definition and template
- [x] Add rspec scenarios
- [x] Generate example files
- [x] Update functional tests to remove welsh language scenarios
- [x] Add usage documentation
- [x] Remove old haml files

### CODE REVIEW

File: engine/config/locales/en.yml [hunk 1]
```diff
@@ -5,6 +5,12 @@ en:
       adviser: Adviser
       example: Example
       important: Important
+    footer:
+      copyright: Copyright ©2021 Citizens Advice. All rights reserved.
```
Comment (davidrapson) [inline engine/config/locales/en.yml:9]: Not worth tackling this here but we should make the year dynamic. Raised https://citizensadvice.atlassian.net/browse/DS-35 to remind us.

File: styleguide/components/footer/footer-docs.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+The constructor accepts the following properties
+
+| Property       | Description                                                |
+| -------------- | ---------------------------------------------------------- |
+| `feedback_url` | If present, a link to this feedback form will be displayed |
+| `homepage_url` | Optional, defaults to `/`                                  |
+
+#### Column slots
+
+As well as the main properties the component also accepts up to for `column` slots.
```
Comment (davidsauntson) [inline styleguide/components/footer/footer-docs.mdx:71]: typo: `for` &rarr; `four`

**Review by davidsauntson — APPROVED** [review]
Lovely stuff! 🦶🏻 

(Shotgun not implementing it on public website... oh).

**Comment by davidrapson** [comment]
> (Shotgun not implementing it on public website... oh).

😅  I did a dry-run for public advice and it's pretty simple. Thankfully the `renders_many` DSL is pretty flexible and supports both passing an array to `columns` or single items to `column`. Ended up with something like the following:

```rb
def public_website_footer_nav_links
  [
    {
      title: I18n.t("footer_links.resources_and_tools_title"),
      links: I18n.t("footer_links.resources_and_tools_links")
    },
    {
      title: I18n.t("footer_links.more_from_us_title"),
      links: I18n.t("footer_links.more_from_us_links")
    },
    ....
  ]
end
```

and:

```haml
= render CitizensAdviceComponents::Footer.new(feedback_url: public_website_feedback_url, homepage_url: homepage_url) do |c|
      = c.columns(public_website_footer_nav_links)
```

**Pull request merged**